scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,304 @@
+        ])
+
+    if 'KUBE_E2E_RUNNER' in os.environ:
+        docker_extra_args.extend([
+          '--entrypoint=%s', os.environ.get('KUBE_E2E_RUNNER')
+        ])
+
+    # exec
+
+    print 'Starting %s...' % CONTAINER
+
+    cmd = ['docker', 'run', '--rm',
+      '--name=%s' % CONTAINER,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    if os.environ.get('JENKINS_GCE_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine:ro'
+            % os.environ['JENKINS_GCE_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_GCE_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine.pub:ro'
+            % os.environ['JENKINS_GCE_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa:ro'
+            % os.environ['JENKINS_AWS_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa.pub:ro'
+            % os.environ['JENKINS_AWS_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_CREDENTIALS_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.aws/credentials:ro'
+            % os.environ['JENKINS_AWS_CREDENTIALS_FILE']])
+
+    if os.environ.get('GOOGLE_APPLICATION_CREDENTIALS'):
+        cmd.extend(['-v', '%s:/service-account.json:ro'
+            % os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+                    '-e', 'GOOGLE_APPLICATION_CREDENTIALS=/service-account.json'])
+
+    for key, value in os.environ.items():
+        if key not in docker_env_ignore:
+            cmd.extend(['-e', '%s=%s' % (key, value)])
+
+    cmd.extend(docker_extra_args)
+
+    cmd.extend([
+      '-e', 'HOME=/workspace',
+      '-e', 'WORKSPACE=/workspace',
+      'gcr.io/k8s-testimages/kubekins-e2e:%s' % e2e_image_tag
+    ])
+
+    try:
+        check(*cmd)
+        print 'Finished Successfully'
+    except subprocess.CalledProcessError as exc:
+        print 'Exiting with code: %d' % exc.returncode # fine?
+
+    signal.signal(signal.SIGTERM, sigterm_handler)
+    signal.signal(signal.SIGINT, sigint_handler)
+
+
+if __name__ == '__main__':
+
+    PARSER = argparse.ArgumentParser(
+        'Runs e2e jobs on the kubernetes repo')
+    PARSER.add_argument(
+        '--branch', default='master', help='Upstream target repo')
+    PARSER.add_argument(
+        '--platform', help='Platform test runs on')
+    PARSER.add_argument(
+        '--env_file', help='Job specific environment file')
+    PARSER.add_argument(
+        '--env_template', help='Job specific environment template')
+    PARSER.add_argument(
+        '--zone', default='us-central1-f', help='Zone for project runs on')
+    PARSER.add_argument(
+        '--project', help='GCP Project Override for the job')
+    PARSER.add_argument(
```

Comment:
Consider yanking out all future functionality except for the minimum required by the flaky job you're testing.